.github/workflows/tag.yml

```diff
@@ -31,11 +32,54 @@ jobs:
         id: create_tag
         uses: mikefarah/yq@v4
         with:
-          cmd: yq -r '"v" + (.MAJ | tostring) + "." + (.MIN | tostring)' ${{ steps.workflow_directory.outputs.name }}/VERSION.yaml
+          cmd: yq -r '"v" + (.MAJ | tostring) + "." + (.MIN | tostring) + "." + (.PATCH | tostring)' ${{ steps.workflow_directory.outputs.name }}/VERSION.yaml
+      - name: Validate tag doesn't already exist
+        run: |
+          if git tag -l | grep -q "${{ steps.create_tag.outputs.result }}"; then
+            echo "Tag ${{ steps.create_tag.outputs.result }} already exists. Exiting."
+            exit 1
+          else
+            echo "Tag ${{ steps.create_tag.outputs.result }} does not exist. Proceeding."
+          fi
+      - name: Validate tag contains all three semantic elements
+        run: |
+          if [[ "${{ steps.create_tag.outputs.result }}" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "Tag ${{ steps.create_tag.outputs.result }} is valid, because it adheres to vMAJ.MIN.PATCH format."
+          else
+            echo "Tag ${{ steps.create_tag.outputs.result }} is not valid. It must adhere to vMAJ.MIN.PATCH format."
+            exit 1
+          fi
+      - name: If this is a PR to a patch branch, ensure the patch version is not 0
+        if: startsWith(github.base_ref, 'patch-')
+        run: |
+          patch_version=$(echo "${{ steps.create_tag.outputs.result }}" | cut -d '.' -f 3)
+          if [ "$patch_version" -ne 0 ]; then
+            echo "Patch version is not 0. This is an acceptable PR to a patch branch."
+          else
+            echo "Patch version is 0. Please ensure that the patch version is incremented before merging this PR."
+            exit 1
+          fi
```

Comment:
Though technically, if `prod` were to be multiple versions (major or minor) behind `dev`, we could theoretically "safely" patch without literally incrementing the patch version ... I think it is clearest + safest to require that patches _do_ increment the patch version.